File: SwarmPortal.Context/ModelInterfaces/ILink.cs

using SwarmPortal.Common;

namespace SwarmPortal.Context;

public interface ILink
{
    ulong Id { get; }
    string Name { get; }
    string Url { get; }
    Group Group { get; }
    bool Enabled { get; }
    IEnumerable<IRole> Roles { get; }
}

File: SwarmPortal.React/src/services/openapi/models/ILink.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { Group } from './Group';
import type { IRole } from './IRole';

export type ILink = {
    readonly id?: number;
    readonly name?: string | null;
    readonly url?: string | null;
    group?: Group;
    readonly enabled?: boolean;
    readonly roles?: Array<IRole> | null;
};

